scripts/monitor-plugin-nuts.js

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env node
+/**
+ * Script to monitor previously launched just-nuts
+ *
+ * args:
+ *   string representation of a jsoan array that contains the data for each pipeline/workflow to monitor
+ *
+ */
+
+const got = require('got');
+const { isArray } = require('@salesforce/ts-types');
+const { sleep, Duration } = require('@salesforce/kit');
+
+const notCompleteStatus = ['running', 'on_hold'];
+
+// circleci
+const circleciBaseUrl = 'https://circleci.com/api/v2/';
+
+const orgSlug = (org, repo) => {
+  return `github/${org}`;
+};
+const projectSlug = (org, repo) => {
+  return `${orgSlug(org)}/${repo}`;
+};
+
+class MonitorPluginNuts {
+  pipelineData = {};
+  isComplete = false;
+  status = 'initial state';
+  justNuts = {};
+  constructor(pipelineData) {
+    this.pipelineData = pipelineData;
+  }
+
+  getWorkflowUrl() {
+    return `${circleciBaseUrl}pipeline/${this.pipelineData.id}/workflow`;
+  }
+
+  getCircleCiUrl() {
+    return `https://app.circleci.com/pipelines/${this.justNuts.project_slug}/${this.justNuts.pipeline_number}/workflows/${this.justNuts.id}`;
+  }
+
+  async circle(url, options = {}) {
+    const defaultOptions = {
+      method: 'GET',
+      headers: { 'Circle-Token': process.env.CIRCLE_TOKEN, responseType: 'json', resolveBodyOnly: true },
+    };
+    options = Object.assign(defaultOptions, { ...options });
+    try {
+      return JSON.parse((await got(url, options)).body);
+    } catch (error) {
+      console.log(error);
+      throw new Error(error && error.response?.body ? error.response.body : error);
+    }
+  }
+
+  async checkWorkflowState() {
+    const url = this.getWorkflowUrl();
+    const response = await this.circle(url, {});
+    const justNuts = response.items.find((item) => item.name === 'just-nuts');
+    if (!justNuts) {
+      this.isComplete = true;
+      return;
+    }
+    this.isComplete = notCompleteStatus.some((status) => justNuts.status !== status);
+    this.justNuts = justNuts;
+  }
+
+  async waitForCompletion() {
+    const retries = 30;
+    let retryCnt = 0;
+    while (!this.isComplete && retryCnt++ <= retries) {
+      await sleep(Duration.seconds(5));
```

Comment:
I set a time limit of 15 minutes to wait for a job to complete
duration needs to be 30. I'll fix.

Reply:
Should retries or sleep duration be exposed as env vars or parameters?